src/components/side-panel/layer-panel/vis-config-slider.js

```diff
@@ -50,31 +85,123 @@ export default function VisConfigSliderFactory(RangeSlider) {
+      }
+    };
+
+    const renderCustomInput = () => {
+      return (
+        <CustomInputWrapper>
+          {isRanged ? (
+            <InputWrapper>
+              <CustomInputLabel>
+                min
+                <RangeInput
+                  type="number"
+                  name="value0"
+                  value={values.value0}
+                  onChange={updateField}
+                  onKeyPress={onKeyPress}
+                />
+              </CustomInputLabel>
+              <CustomInputLabel>
+                max
+                <RangeInput
+                  type="number"
+                  name="value1"
+                  value={values.value1}
+                  onChange={updateField}
+                  onKeyPress={onKeyPress}
+                />
+              </CustomInputLabel>
+            </InputWrapper>
           ) : (
-            <FormattedMessage id={`property.${property}`} />
+            <InputWrapper>
+              <RangeInput
+                type="number"
+                name="value"
+                value={values.value}
+                onChange={updateField}
+                onKeyPress={onKeyPress}
+              />
+            </InputWrapper>
           )}
-        </PanelLabel>
-      ) : null}
-      <RangeSlider
-        range={range}
-        value0={isRanged ? config.visConfig[property][0] : range[0]}
-        value1={isRanged ? config.visConfig[property][1] : config.visConfig[property]}
-        step={step}
-        isRanged={Boolean(isRanged)}
-        onChange={value => onChange({[property]: isRanged ? value : value[1]})}
-        inputTheme={inputTheme}
-        showInput
-      />
-    </SidePanelSection>
-  );
+        </CustomInputWrapper>
+      );
+    };
+
+    return (
+      <SidePanelSection disabled={Boolean(disabled)}>
+        {label ? (
+          <PanelLabel>
+            {typeof label === 'string' ? (
+              <FormattedMessage id={label} />
+            ) : typeof label === 'function' ? (
+              <FormattedMessage id={label(config)} />
+            ) : (
+              <FormattedMessage id={`property.${property}`} />
+            )}
+          </PanelLabel>
+        ) : null}
+
+        <InputWrapper>
+          <CustomInputLabel>custom input</CustomInputLabel>
+          <Checkbox
+            id={`property.${property}`}
+            checked={custom}
+            onChange={() => setCustom(!custom)}
+          />
+        </InputWrapper>
+
+        {!custom ? (
+          <RangeSlider
+            range={range}
+            value0={isRanged ? config.visConfig[property][0] : range[0]}
+            value1={isRanged ? config.visConfig[property][1] : config.visConfig[property]}
+            step={step}
+            isRanged={Boolean(isRanged)}
+            onChange={value => onChange({[property]: isRanged ? value : value[1]})}
+            inputTheme={inputTheme}
+            showInput
+          />
+        ) : (
+          renderCustomInput()
```

Comment:
now we can just render our `CustomInput` component passing in the value from prop
```
        <CustomInput
            isRanged={isRanged}
            value={config.visConfig[property]}
            onChangeCustomInput={v => onChange({[property]: v})}
          />
```